### PR TITLE
Fix the bug that deployment timeout error was not handled

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/controller/scheduler.go [hunk 1]
```diff
@@ -327,27 +327,30 @@ func (s *scheduler) Run(ctx context.Context) error {
...
-		if sigType == executor.StopSignalCancel {
+		if result == model.StageStatus_STAGE_CANCELLED {
 			deploymentStatus = model.DeploymentStatus_DEPLOYMENT_CANCELLED
-			statusReason = fmt.Sprintf("Deployment was cancelled by %s while executing stage %s", cancelCommander, ps.Id)
+			statusReason = fmt.Sprintf("Cancelled by %s while executing stage %s", cancelCommander, ps.Id)
 			break
 		}
 
-		// The stage was failed but not caused by the stop signal.
-		if result == model.StageStatus_STAGE_FAILURE && sigType == executor.StopSignalNone {
```
Comment (nghialv) [inline pkg/app/piped/controller/scheduler.go:345]: The original code was missing the handling of the timeout error.

File: pkg/app/piped/controller/scheduler.go [hunk 2]
```diff
@@ -327,27 +327,30 @@ func (s *scheduler) Run(ctx context.Context) error {
 		}
 
 		// If all operations of the stage were completed successfully
-		// go the next stage to handle.
+		// forward to handle the next stage.
```
Comment (khanhtc1202) [inline pkg/app/piped/controller/scheduler.go:330]: nit, ~forward to~ handle the next stage.

File: pkg/app/piped/controller/scheduler.go [hunk 3]
```diff
@@ -220,7 +220,7 @@ func (s *scheduler) Run(ctx context.Context) error {
 	repoCfg, ok := s.pipedConfig.GetRepository(repoID)
 	if !ok {
 		s.doneDeploymentStatus = model.DeploymentStatus_DEPLOYMENT_FAILURE
-		statusReason = fmt.Sprintf("Unable to find %q from the repository list in piped config", repoID)
+		statusReason = fmt.Sprintf("Repository %q is not found in the piped config", repoID)
```
Comment (khanhtc1202) [inline pkg/app/piped/controller/scheduler.go:223]: nit, `statusDesc` or `statusDescription` can be considered as a better name 🤔

Comment (nghialv) [reply pkg/app/piped/controller/scheduler.go:223]: That name was already fixed/used in the proto so let me keep it as is.

Comment (khanhtc1202) [reply pkg/app/piped/controller/scheduler.go:223]: 🙆‍♀️

File: pkg/app/piped/controller/scheduler.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ import (
 )
 
 var (
-	defaultDeploymentTimeout = time.Hour
+	defaultDeploymentTimeout = 6 * time.Hour
```
Comment (khanhtc1202) [inline pkg/app/piped/controller/scheduler.go:39]: nit, do we have docs that related to this default value 👀

Comment (nghialv) [reply pkg/app/piped/controller/scheduler.go:39]: This will be included while fixing https://github.com/pipe-cd/pipe/issues/1373

Comment (khanhtc1202) [reply pkg/app/piped/controller/scheduler.go:39]: got it, thank you 👍

**Review by nakabonne — APPROVED** [review]
👍

**Pull request merged**